File: Season 2 Leetcode/Easy/136-SingleNumber.java
package Easy;

class Solution {
    public int singleNumber(int[] nums) {
        // Here we use XOR (Bitwise)
        // This is because when you have let's say [2,2,1]
        // 1st pass:
        // result = 0
        // 0 XOR 2 = 2
        // result -> 2
        // 2nd pass:
        // result = 2
        // 2 XOR 2 = 0
        // result -> 0
        // 3rd pass:
        // result = 0
        // 0 XOR 1 = 0
        // result -> 1
        int result = 0;
        for (int num : nums ) {
         	result ^= num;   
        }
	    return result;
    }
}

File: 2023/Arrays/Easy/136-SingleNumber.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        my_dict = {}
        for num in nums:
            if num not in my_dict:
                my_dict[num] = 0
            my_dict[num] += 1

        for key in my_dict:
            if my_dict[key] == 1:
                return key

        return 0